c++/src/kj/async.h

```diff
@@ -458,6 +458,21 @@ PromiseForResult<Func, void> evalLast(Func&& func) KJ_WARN_UNUSED_RESULT;
 // callback enqueues new events, then latter callbacks will not execute until those events are
 // drained.
 
+template <typename E, typename T, typename... Args>
+kj::Promise<T> runInEnvironment(E&& environment, FunctionParam<kj::Promise<T>(Args...)> func);
```

Comment:
Since this is a template, there's no need for `FunctionParam`. The second param can just be `Func&& func`, where `Func` is a template parameter.

This has two benefits:
1. No need to manually cast to `FunctionParam` at call sites to make template matching work.
2. Avoids forcing the callbacks to go through a virtual function invocation.

Note when you invoke `evalNow` in the implementation, it should be `evalNow(kj::fwd<Func>(func))`.